### PR TITLE
Few fixes & font/translation improvements

### PR DESCRIPTION
**Pull request by Ghabry**
First commit fixes Comic Sans on Windows ;). Reported by Mimigris:musical_note:. Thanks!

Second commit fixes language selection overflowing the screen because YNOnline has more than 9 translations. WOW. Reported by Mimigris:musical_note:. Thanks!

Third commit fixes an oversight/regression in the rendering order: When on the same Y-coordinate the Player is always drawn first, independent of the X-coordinate. Found in the game Collective Unconcious and reported by UliZ. Thanks!

### CODE REVIEW

**Comment by Ghabry** [comment]
More minor things I consider adding here:

- "Language" text cannot be translated
- A way to specify that a translation shall not use the provided game Font but the builtin font (e.g. game is Japanese with a Font.ttf and loading a Russian translation will look bad because cyrillic is usually full width in these fonts)
- Degrade the Freetype glyph errors to debug and fallback to builtin if they happen. Some fonts seem to be bogus and trigger these errors.

**Comment by Ghabry** [comment]
The ``Meta.ini`` learned new fields:

- ``Term=`` for specifying the Term for "Language" shown in the menu
- ``Font=Builtin`` (anything that is not "Builtin" is currently ignored) disables a Font(2) provided by the game and uses our builtin bitmap font. This is useful when the Font of the game has unsuitable glyphs for the new language and the translation does not want to provide a custom Font.

A ``language/default/Meta.ini`` allows providing custom text for the default game language (the first item in the language list). Only the keys ``Name=``, ``Description=`` and ``Term=`` are handled. Everything else is ignored. No asset redirection happens.

-----------

The guide will need an update before 0.7.1. :thinking:

**Review by fdelapena — APPROVED** [review]

**Comment by Ghabry** [comment]
Not creating an issue for this: There was a report in chat that battle test in the editor is broken. Some refactor of the command line parser must have lost the BattleTest edge case: The command line is usually ``BattleTest ShowTitle Window 1`` and this failed to parse because the Troop ID is not directly after ``BattleTest``.

I added a special case now when the first argument is ``BattleTest`` it does it in a RPG_RT compatible way. Otherwise it checks the arguments after. ``--battle-test`` is unaffected and always reads the numbers right after.

My code also fixes corner cases like: "--battle-test 1 Window". Here "Window" argument was _eaten_ because battle-test reads either 1 or 4 arguments depending on engine version. This is now detected and the parser rewinds onto "Window".

```cpp
diff --git a/src/cmdline_parser.cpp b/src/cmdline_parser.cpp
index c781421d..37052643 100644
--- a/src/cmdline_parser.cpp
+++ b/src/cmdline_parser.cpp
@@ -131,3 +131,8 @@ void CmdlineParser::SkipNext() {
 	++index;
 }
 
+void CmdlineParser::RewindBy(int rewind) {
+	assert(rewind >= 0);
+	index -= rewind;
+	assert(index >= 0);
+}
diff --git a/src/cmdline_parser.h b/src/cmdline_parser.h
index 2dbb2e61..bd7ace97 100644
--- a/src/cmdline_parser.h
+++ b/src/cmdline_parser.h
@@ -136,6 +136,12 @@ public:
 
 	/** Resets the parser back to the first argument */
 	void Rewind();
+
+	/**
+	 * Rewinds the parser by the specified amount
+	 * @param rewind how far to rewind
+	 */
+	void RewindBy(int rewind);
 private:
 	template <typename F>
 		bool DoParseNext(CmdlineArg& arg, int max_values, F&& is_longval, char shortval);
diff --git a/src/player.cpp b/src/player.cpp
index b05d790c..bd52771b 100644
--- a/src/player.cpp
+++ b/src/player.cpp
@@ -477,6 +477,46 @@ Game_Config Player::ParseCommandLine(int argc, char *argv[]) {
 
 	auto cfg = Game_Config::Create(cp);
 
+	bool battletest_handled = false;
+
+	cp.Rewind();
+	if (!cp.Done()) {
+		// BattleTest argument handling in a RPG_RT compatible way is very ugly because the arguments to not follow
+		// directly. Try to parse it and afterwards rewind the parser to parse the rest.
+		CmdlineArg arg;
+		long li_value = 0;
+
+		// Legacy BattleTest handling: When BattleTest is argument 0, then the values are:
+		// - arg3: troop_id
+		// - arg4-7: formation, condition, terrain_id (2k3 only)
+		// - arg1-2 are unrelated ("ShowTitle Window")
+		if (cp.ParseNext(arg, 6, {"battletest"})) {
+			Game_Battle::battle_test.enabled = true;
+			Game_Battle::battle_test.troop_id = 0;
+
+			// Starting from 3 to reach arg3 from arg0
+			if (arg.NumValues() >= 3) {
+				if (arg.ParseValue(2, li_value)) {
+					Game_Battle::battle_test.troop_id = li_value;
+				}
+			}
+
+			if (arg.NumValues() >= 6) {
+				if (arg.ParseValue(3, li_value)) {
+					Game_Battle::battle_test.formation = static_cast<lcf::rpg::System::BattleFormation>(li_value);
+				}
+				if (arg.ParseValue(4, li_value)) {
+					Game_Battle::battle_test.condition = static_cast<lcf::rpg::System::BattleCondition>(li_value);
+				}
+				if (arg.ParseValue(5, li_value)) {
+					Game_Battle::battle_test.terrain_id = static_cast<lcf::rpg::System::BattleFormation>(li_value);
+				}
+			}
+
+			battletest_handled = true;
+		}
+	}
+
 	cp.Rewind();
 	while (!cp.Done()) {
 		CmdlineArg arg;
@@ -504,7 +544,7 @@ Game_Config Player::ParseCommandLine(int argc, char *argv[]) {
 			hide_title_flag = true;
 			continue;
 		}
-		if(cp.ParseNext(arg, 4, {"battletest", "--battle-test"})) {
+		if(!battletest_handled && cp.ParseNext(arg, 4, {"battletest", "--battle-test"})) {
 			// Legacy RPG_RT argument - battletest
 			Game_Battle::battle_test.enabled = true;
 			Game_Battle::battle_test.troop_id = 0;
@@ -519,13 +559,22 @@ Game_Config Player::ParseCommandLine(int argc, char *argv[]) {
 				// 2k3 passes formation, condition and terrain_id as args 5-7
 				if (arg.ParseValue(1, li_value)) {
 					Game_Battle::battle_test.formation = static_cast<lcf::rpg::System::BattleFormation>(li_value);
+				} else {
+					// When the argument is not a number, args5-7 were likely not specified and are something different
+					// Rewind to prevent losing other args
+					cp.RewindBy(3);
+					continue;
 				}
+
 				if (arg.ParseValue(2, li_value)) {
 					Game_Battle::battle_test.condition = static_cast<lcf::rpg::System::BattleCondition>(li_value);
 				}
 				if (arg.ParseValue(3, li_value)) {
 					Game_Battle::battle_test.terrain_id = static_cast<lcf::rpg::System::BattleFormation>(li_value);
 				}
+			} else if (arg.NumValues() >= 2) {
+				// Only troop-id was provided: Rewind to prevent losing other args
+				cp.RewindBy(arg.NumValues() - 1);
 			}
 			continue;
 		}
```

**Comment by Ghabry** [comment]
I think this is everything :)
